x-pack/legacy/plugins/alerting/server/alerts_client.test.ts

```diff
@@ -727,6 +737,216 @@ describe('disable()', () => {
   });
 });
 
+describe('muteAll()', () => {
+  test('mutes an alert', async () => {
+    const alertsClient = new AlertsClient(alertsClientParams);
+    savedObjectsClient.get.mockResolvedValueOnce({
+      id: '1',
+      type: 'alert',
+      attributes: {
+        muteAll: false,
+      },
+      references: [],
+    });
+
+    await alertsClient.muteAll({ id: '1' });
+    expect(savedObjectsClient.update).toHaveBeenCalledWith(
+      'alert',
+      '1',
+      { muteAll: true, mutedInstanceIds: [], updatedBy: 'elastic' },
+      { references: [] }
+    );
+  });
+
+  test('skips muting when alert already muted', async () => {
```

Comment:
Is there room to consider not skipping but rather making it explicit that the action failed and why?
Obviously if in other places we skip then we should keep the behaviour uniform, but I'm not a fan of No-Ops tha appear to be successful as they can often obscure an underlying problem (such as UI not updating relative to retrieved data etc.)

Reply:
Yeah this is to keep it uniform with the design of enable / disable which also skips if ever the alert is already in that state. Returning error isn't a bad idea but I guess it comes down to what behaviour we prefer our APIs to have.

Reply:
Fair.
I guess, broadly, I tend to bias for explicitness and would prefer that we did return an error, but definitely only if it was done that way in all cases, and definitely isn't a blocker.

Reply:
Just moving @bmcconaghy's comment (https://github.com/elastic/kibana/pull/43712#discussion_r329385502) here:

> To me if I call an API to put something a state, and it's already in that state, that's not an error, so it makes sense to just say "OK" in that case. What value would returning an error response give? The client would then have to figure out what state things are in, and oh yeah, they're in the state the client wanted anyway. Seems like a distinction without a difference to me.